pkg/controller/statefulset/stateful_set.go

```diff
@@ -319,12 +321,9 @@ func (ssc *StatefulSetController) adoptOrphanRevisions(set *apps.StatefulSet) er
 		}
 	}
 	if hasOrphans {
-		fresh, err := ssc.kubeClient.AppsV1().StatefulSets(set.Namespace).Get(set.Name, metav1.GetOptions{})
-		if err != nil {
-			return err
-		}
-		if fresh.UID != set.UID {
-			return fmt.Errorf("original StatefulSet %v/%v is gone: got uid %v, wanted %v", set.Namespace, set.Name, fresh.UID, set.UID)
+		canAdoptErr := ssc.canAdoptFunc(set)()
+		if canAdoptErr != nil {
+			return fmt.Errorf("can't adopt ControllerRevisions: %v", canAdoptErr)
```

Comment:
StatefulSet does not use `ControllerRevisionControllerRefManager` [like DaemonSet](https://github.com/kubernetes/kubernetes/blob/59aa8fd99e2e0c566310492185658d51f6343782/pkg/controller/daemon/update.go#L253-L267) to manage controller revisions, so this is quick.

Maybe we can migrate to use ControllerRevisionControllerRefManager in the future?

Reply:
That sounds like a good idea. Can you create an issue on this?

Reply:
sure, https://github.com/kubernetes/kubernetes/issues/88150